framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1187,6 +1213,18 @@ public AnnotatedTypeMirror fromElement(Element elt) {
                             + elt);
         }
 
+        if (checker.hasOption("mergeStubsWithSource")) {
+            AnnotatedTypeMirror stubType = stubTypes.getAnnotatedTypeMirror(elt);
+            if (stubType != null) {
+                if (type == null) {
+                    type = stubType;
+                } else {
+                    // Must merge (rather than only take the stub type if it is a subtype)
```

Comment:
It does not. What it does do is prefer the types from stub files. For WPI, this is fine (and has the same effect as GLB), because the stub types include any types the user wrote in the program.

I tried writing a version of `AnnotatedTypesMerger` that actually does GLB, but I'm struggling to get it to pass the WPI tests when I substitute it here. You can find it on the `glb-failure` branch of my fork. I'm not sure what's going wrong - the replacement seems to be happening correctly, but the ATF retains incorrect types and fails the stub WPI tests with ~35 errors. Any insight you have would be appreciated.

If we can't figure out what's wrong there, we might need to change the documentation of the `-AmergeStubsWithSource` feature. It's unclear to me who else besides WPI users would be interested in it, so we don't have to strictly hold to the contract we've written into those docs - I'd prefer to keep the working version we have here that replaces rather than merges than to spend a lot more time trying to figure out how to merge.